src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { memo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoInfo } from '../TodoInfo/TodoInfo';
+
+type Props = {
+  todos: Todo[];
+  onDeleteTodo: (todoId: number) => void;
+  tempTodo: Todo | null;
+  isAdding: boolean;
+  onUpdateTodo: (
+    todoId: number,
+    fieldsToUpdate: Partial<Pick<Todo, 'title' | 'completed'>>
+  ) => void;
+  updatingTodosIds: number[];
+};
+
+export const TodoList: React.FC<Props> = memo(
+  ({
+    todos,
+    onDeleteTodo,
+    tempTodo,
+    isAdding,
+    onUpdateTodo,
+    updatingTodosIds,
+  }) => {
+    return (
+      <section className="todoapp__main" data-cy="TodoList">
+        {todos.map((todo) => (
+          <TodoInfo
+            todo={todo}
+            onDeleteTodo={onDeleteTodo}
+            onUpdateTodo={onUpdateTodo}
+            shouldShowLoader={updatingTodosIds.includes(todo.id)}
+          />
+        ))}
+
+        {tempTodo && (
+          <TodoInfo
+            todo={tempTodo}
+            onDeleteTodo={onDeleteTodo}
+            isAdding={isAdding}
+            onUpdateTodo={onUpdateTodo}
+            shouldShowLoader={updatingTodosIds.includes(tempTodo.id)}
```

Comment:
shouldShowLoader for tempTodo should always be true, so this check is redundant.
```suggestion
            shouldShowLoader
```

Reply:
Yeap, I agree with @oleh-yablunovskyi